beacon_node/operation_pool/src/attestation_storage.rs

```diff
@@ -172,19 +331,101 @@ impl<E: EthSpec> AttestationMap<E> {
         }
     }
 
+    /// Aggregate Electra attestations for the same attestation data signed by different
+    /// committees.
+    ///
+    /// Non-Electra attestations are left as-is.
+    pub fn aggregate_across_committees(&mut self, checkpoint_key: CheckpointKey) {
```

Comment:
@michaelsproul why is it necessary to pre-aggregate across committees before running max-cover? All committees are exclusive so you can run max-cover on each committee, pick the collection of solutions and merge them. Is there any efficiency reason? Seems simpler to merge the end instead of mutating the pool every run.

Reply:
Response: The final selection of attestations can only include 8 distinct attestation data. Running max-cover first on each committee index may not yield a set of results that have overlapping attestation data. Aggregating across committees makes the max-cover algo aware of this limitation and produces better results.